### ISSUE TITLE
mqtt messages are wrongly duplicated in the output of a mqtt in node

### ISSUE THREAD

**Issue by hsmptg** [node-red/node-red#4385]
### Current Behavior

If I use a "mqtt in" node with a topic like "manusa/power" and connect it to a "debug" node, when I receive that message I see it in the debug messages as expected. But if I add another "mqtt in" node this time with topic "manusa/#" I get a duplicated debug message even without this second "mqtt in" node being connected to the "debug" node!?!

### Expected Behavior

Since the second "mqtt in" is not connected with the debug node, I would expect that its messages would be displayed in the debug messages!

### Steps To Reproduce

![imagem](https://github.com/node-red/node-red/assets/6853926/c4800930-e6f1-4ad3-ae0d-2cfde4eb7a0a)

### Example flow

Please see the above image.

### Environment

- Node-RED version: 3.1.0
- Node.js version: 16.20.2
- npm version: 8.19.4
- Platform/OS: Linux
- Browser: Firefox

**Comment by hsmptg** [comment]
Please notice that the "manusa/power" "mqtt in" node generates 2 output messages even when the broker sends a single message!
If I delete the "manusa/#" "mqtt in" node I get the single message as expected!
The initial behaviour would be OK if I had a connection between this last "mqtt in" node and the debug node, that I do not have!

**Comment by knolleary** [comment]
This is a limitation of MQTT 3.1 and certain broker implementations.

When you have subscriptions that 'overlap', some brokers send one messages, some send two. The protocol spec allows for either behaviour.

When those messages arrive in Node-RED, we only know the topic it was published to, not which subscription triggered it. So we have to match it against all MQTT nodes to find ones that match.

As the broker is sending two messages for this scenario, they both match both nodes so both send on that message.

You could try modifying the broker config node to use MQTTv5 instead. That version of the protocol allows us to correlate each message with the subscription that triggered it and should solve this issue.

**Comment by Steve-Mcl** [comment]
This has been raised previously and it is not something we can handle due partly to the way MQTT is designed and the way it is abstracted into an easy to use Node-RED node.

See this comment for details: https://github.com/node-red/node-red/issues/4131#issuecomment-1501909649

If you switch to MQTT v5 (on the MQTT config) and your broker supports v5, we attempt to get around the limitation by including a subscription identifier in the transmission meaning we can identify which node subscribed.

In short, your options are:
1. Dont subscribe to same topic twice
2. Use MQTT v5 and a suitable broker.

**Comment by hsmptg** [comment]
Thanks for the replies but I am not sure if we are talking about the same thing.
In the [#4131 (comment)](https://github.com/node-red/node-red/issues/4131) its mentioned that one of the messages is retained, something that I do have here.
Besides if I use the following instead I do not get the duplicate issue:
![imagem](https://github.com/node-red/node-red/assets/6853926/873e127a-60af-43d3-a0e1-b771225a34cd)
The problem only appear if I have a wildcard subscription, like manusa/#.

**Comment by hsmptg** [comment]
Nevertheless I can confirm that choosing MQTT V5 protocol in Node-RED (something that my EMQX broker allows) I do not get the error I initially reported.
Thanks for all your help!!!

**Comment by Steve-Mcl** [comment]
> In the [#4131 (comment)](https://github.com/node-red/node-red/issues/4131) its mentioned that one of the messages is retained, something that I do have here.

Yeah, the subject matter was different but the underlying issue was the same.

Happy you are sorted.
